### DISCUSSION TITLE
Allow to set test run type as configuration

### DISCUSSION THREAD

**Discussion by MGarfOppLoans** [forcedotcom/salesforcedx-vscode#5796] [Ideas]
**Is your feature request related to a problem? Please describe.**
Currently when you click run tests it uses the default test run type which can be long running for test development

**Describe the solution you'd like**
Allow configuration to set the run type so my team can default to Tap for faster test runs when developing and back to the default when debugging. Currently we rely on aliases and vscode tasks to attempt the same outcome but it would be much cleaner if we can in settings set with test run type to run

See hardcoded reference to human on: 

`\packages\salesforcedx-vscode-apex\src\commands\forceApexTestRun.ts`

line 143: `.withFlag('--resultformat', 'human')`

can be replaced with:

`.withFlag('--resultformat', vscode.workspace.getConfiguration('testFormat') ? vscode.workspace.getConfiguration('testFormat') | 'human');`

**Comment by lcampos** [comment]
Hi @MGarfOppLoans , thanks for opening this suggestion. Independent to the result format, have you or someone else on your team tried using the `salesforcedx-vscode-core.experimental.useApexLibrary` setting ? We've been working on performance enhancements for running Apex tests and wonder if your experience with the new code would be better.

**Comment by MGarfOppLoans** [comment]
Thanks for the tip I'll give it a shot

**Comment by MGarfOppLoans** [comment]
turns out i had that on, tap seems to still be faster but did notice some speed improvements with the apex library
